### PR TITLE
chore: Show scrollbars in selected integration tests

### PR DESCRIPTION
**Pull request by jperals**
### Description

Ticket: `AWSUI-59983`

- Removed the `--hide-scrollbars` flag from browser-test-utils to see which tests would fail: see [failing run](https://github.com/cloudscape-design/browser-test-tools/actions/runs/11889829835/job/34192685449)
- In this PR, all those failing tests are modified to use scrollbars and fixed as necessary

For each case, either made small modifications to the test conditions to avoid scrollbars (e.g, viewport size or remove elements), or just changed measurements to account for the scrollbars, if the former option was complicating the testing in any way.

### How has this been tested?

The integration tests themselves

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/__integ__/use-browser-with-scrollbars.ts [hunk 1]
```diff
@@ -4,63 +4,83 @@ import useBrowser from '@cloudscape-design/browser-test-tools/use-browser';
...
+        args: flags,
+      },
+    },
+    width,
+    height,
+  };
   return useBrowser(options, testFn);
 }
+
+export const scrollbarThickness = 15;
```
Comment (jperals) [inline src/__integ__/use-browser-with-scrollbars.ts:84]: When this file is removed, this const can go somewhere else. At the moment, this util is what it is most closely related to.

File: pages/date-picker/positioning.page.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ export default function DatePickerScenario() {
         style={{
           inlineSize: '180px',
           marginInlineStart: '300px',
-          marginBlockStart: '300px',
+          marginBlockStart: '285px',
```
Comment (jperals) [inline pages/date-picker/positioning.page.tsx:17]: The page content overflows the window horizontally, which leads to a horizontal scrollbar appearing at the bottom of the page, taking up 15px of vertical space and altering the test results.

Verified that this page is not used for visual regression.

File: src/app-layout/__integ__/mobile.test.ts [hunk 3]
```diff
@@ -50,7 +50,7 @@ describe.each(['classic', 'visual-refresh', 'visual-refresh-toolbar'] as const)(
       setupTest(async page => {
         await page.visit(pageName);
         const [documentWidth, windowWidth] = await page.getViewportWidths();
-        expect(documentWidth).toEqual(windowWidth);
+        expect(documentWidth).toBeLessThanOrEqual(windowWidth);
```
Comment (jperals) [inline src/app-layout/__integ__/mobile.test.ts:53]: The dev page has a page-level scrollbar which makes the inner document element be 15px smaller than the window.

Changing from `toEqual` to `toBeLessThanOrEqual` preserves the purpose of the test, which is to verify that the document is _not wider_ than the page (and thus overflowing horizontally).

File: pages/app-layout/runtime-drawers-with-updates.page.tsx [hunk 4]
```diff
@@ -128,6 +128,7 @@ export default function WithDrawers() {
       tools={<Info helpPathSlug={helpPathSlug} />}
       toolsOpen={isToolsOpen}
       toolsHide={!hasTools}
+      navigationHide={true}
```
Comment (abdhalees) [inline pages/app-layout/runtime-drawers-with-updates.page.tsx:131]: why is this needed?

Comment (jperals) [reply pages/app-layout/runtime-drawers-with-updates.page.tsx:131]: The inner content width slightly shrinking because of the vertical scrollbar was affecting the test, and since the side navigation in this dev page is empty (nothing passed to the `navigation` slot) and not relevant for any test performed on this dev page (also visual regression tests), hiding it to free space was more convenient and less reliant on magic numbers than accommodating the viewport width or other measurements.

**Review by abdhalees — APPROVED** [review]
